components/sharing-dialog/src/helpers/helpers.js

```diff
@@ -0,0 +1,249 @@
+import {
+    ACCESS_VIEW_ONLY,
+    ACCESS_NONE,
+    ACCESS_VIEW_AND_EDIT,
+    SHARE_TARGET_EXTERNAL,
+    SHARE_TARGET_PUBLIC,
+} from '../sharing-constants.js'
+
+/**
+ * Returns a function, that, as long as it continues to be invoked, will not be triggered. The
+ * function will be called after it stops being called for N milliseconds. If `immediate` is
+ * passed, trigger the function on the leading edge, instead of the trailing.
+ */
+
+export const debounce = (func, wait, immediate) => {
+    let timeout
+
+    return (...args) => {
+        const context = this
+
+        const later = () => {
+            timeout = null
+
+            if (!immediate) {
+                func.apply(context, args)
+            }
+        }
+
+        const callNow = immediate && !timeout
+
+        clearTimeout(timeout)
+        timeout = setTimeout(later, wait)
+
+        if (callNow) {
+            func.apply(context, args)
+        }
+    }
+}
+
+/**
+ * Name conversion
+ */
+
+export const nameToInitials = (name) => {
+    if (!name) {
+        return ''
+    }
+
+    const parts = name.split(' ')
+    const firstNameInitial = parts.length > 0 ? parts.shift().charAt(0) : ''
+    const lastNameInitial = parts.length > 0 ? parts.pop().charAt(0) : ''
+
+    return `${firstNameInitial}${lastNameInitial}`
+}
+
+/**
+ * Access and constant conversion
+ */
+
+export const convertAccessToConstant = (access) => {
+    if (typeof access === 'boolean') {
+        return access ? ACCESS_VIEW_ONLY : ACCESS_NONE
+    }
+
+    if (/^rw/.test(access)) {
+        return ACCESS_VIEW_AND_EDIT
+    } else if (/^r-/.test(access)) {
+        return ACCESS_VIEW_ONLY
+    } else {
+        return ACCESS_NONE
+    }
+}
+
+export const convertConstantToAccess = (constant, useBoolean) => {
+    switch (constant) {
+        case ACCESS_NONE:
+            return useBoolean ? false : '--------'
+        case ACCESS_VIEW_ONLY:
+            return useBoolean ? true : 'r-------'
+        case ACCESS_VIEW_AND_EDIT:
+            return useBoolean ? true : 'rw------'
+        default:
+            return useBoolean ? false : '--------'
+    }
+}
+
+/**
+ * Mapping users and groups to style used internally
+ */
+
+export const mapUsers = ({ id, name, access }) => ({
+    id,
+    name,
+    access: convertAccessToConstant(access),
+})
+
+export const mapGroups = ({ id, name, access }) => ({
+    id,
+    name,
+    access: convertAccessToConstant(access),
+})
+
+/**
+ * Helper to check whether to allow removing for the selected target
+ */
+
+export const isRemovableTarget = (target) => {
+    const permanentTargets = [SHARE_TARGET_EXTERNAL, SHARE_TARGET_PUBLIC]
+
+    // Do not allow removal of permanent targets
+    return !permanentTargets.includes(target)
+}
```

Comment:
Just a suggestions
```suggestion
const permanentTargets = [SHARE_TARGET_EXTERNAL, SHARE_TARGET_PUBLIC]
export const isRemovableTarget = (target) => {
    // Do not allow removal of permanent targets
    return !permanentTargets.includes(target)
}
```

Reply:
Yeah sounds good, fixed in 1e77401dd9815696c812f88cd36ba9a2b0d9aa5a.